client/src/app/beacons/[id]/page.tsx

```diff
@@ -0,0 +1,39 @@
+"use client";
+import { useGetBeaconQuery } from "@/redux/api";
+import { usePathname, useRouter } from "next/navigation";
+import { useEffect, useState } from "react";
+import { toast } from "react-toastify";
+import { useTheme } from "next-themes";
+import BeaconDetailsPageTemplate from "@/components/templates/beacon-deatails.template";
+
+const BeaconDetailsPage = () => {
+    const path = usePathname();
+    const router = useRouter();
+    const [beaconId, setBeaconId] = useState<string | undefined>(undefined);
+    const { data: beacon, error, isLoading } = useGetBeaconQuery(beaconId ?? "", {
+        skip: !beaconId,
+    });
+
+    useEffect(() => {
+        const pathArr = path.split("/");
+        if (pathArr.length < 3) {
```

Comment:
there's a better way to get the dynamic segment from the path https://nextjs.org/docs/pages/building-your-application/routing/dynamic-routes